addons/portal/controllers/mail.py

```diff
@@ -111,12 +111,22 @@ def _portal_post_has_content(self, thread_model, thread_id, message, attachment_
         """ Tells if we can effectively post on the model based on content. """
         return bool(message) or bool(attachment_ids)
 
-    @http.route('/mail/avatar/mail.message/<int:res_id>/author_avatar/<int:width>x<int:height>', type='http', auth='public')
-    def portal_avatar(self, res_id=None, height=50, width=50, access_token=None, _hash=None, pid=None):
+    @http.route('/mail/avatar/mail.message/<int:message_id>/<int:thread_id>/<thread_model>/author_avatar/<int:width>x<int:height>', type='http', auth='public')
```

Comment:
We usually have model before id in binary routes (I would put them even before the message id in this case).